OPSEXP-1821 Update helm docs with Rancher Desktop deployment (#1003)
Alfresco/acs-deployment: docs/helm/desktop-deployment.md

> # Alfresco Content Services Helm Deployment with Desktop

Reviewer: ```suggestion
# Alfresco Content Services Helm Deployment on local machines
```

This page describes how to deploy Alfresco Content Services (ACS) Enterprise or Community using [Helm](https://helm.sh) onto [Rancher Desktop](https://rancherdesktop.io/) and  [Docker for Desktop](https://docs.docker.com/desktop/).

## Prerequisites

- You've read the projects [main README](/README.md#prerequisites) prerequisites section
- You've read the [main Helm README](./README.md) page
- You are proficient in Kubernetes
- A machine with at least 16GB memory

> - [Rancher for Desktop](https://rancherdesktop.io/). Rancher Desktop includes kubectl and Helm as pre-installed tools, ready to use right after installation.

Reviewer: ```suggestion

Having installed one between:

- [Rancher for Desktop](https://rancherdesktop.io/). Includes kubectl and Helm, ready to use right after installation.
- [Docker for Desktop](https://docs.docker.com/desktop/). Requires separate install of kubectl and Helm.
```

### Rancher Desktop specific configuration

Uncheck `Enable Traefik` from the `Kubernetes Settings` page to disable Traefik. You may need to exit and restart Rancher Desktop for the change to take effect. Ref: [Setup NGINX Ingress Controller](https://docs.rancherdesktop.io/how-to-guides/setup-NGINX-Ingress-Controller)

### Docker Desktop specific configuration

[Prerequisites](./desktop-deployment.md#Prerequisites) are consistent with those for Docker Desktop. Additionally, it is essential to install the latest version of [Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl) & [Helm](https://helm.sh/docs/intro/install).

After the installation of Docker Desktop, the following configurations should be adjusted within Docker Desktop settings.

> `Settings > Resources > Advanced > CPUs:8, Memory: 16GB, Swap: 1GB`
> `Settings > kubernetes > Enable Kubernetes`

Reviewer: ```suggestion
* `Settings > Resources > Advanced > CPUs:8, Memory: 16GB, Swap: 1GB`
* `Settings > kubernetes > Enable Kubernetes`
```

better use bullets

After making the necessary settings `Apply and restart` the docker desktop.

## Deployment

After successfully launching Rancher Desktop, proceed with the instructions provided in the following sections to install ACS (Enterprise or Community) on your local system.

> > Note: To execute the deployment on Docker Desktop, please adhere to the same set of deployment instructions provided below.

Reviewer: do not reference Rancher or Docker desktop from here onwards, use a generic reference like local machine or similar

### Namespace

To establish an isolated environment for ACS within the Kubernetes cluster, initiate the creation of a Kubernetes namespace using the provided command. Throughout the subsequent sections of this tutorial, we will consistently refer to this namespace as 'alfresco'

```bash
kubectl create namespace alfresco
```

### Ingress

> Utilize the subsequent command to include the chart repository:

Reviewer: I don't like unnecessary wording before a command, I will prefer for readability:

```suggestion
Add the ingress-nginx chart repository:
```

it's already clear for me that it's a command because we are using code fences

```bash
helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
helm repo update
```

> Execute the following command to install an ingress controller within the 'alfresco' namespace:

Reviewer: same above

# Verify NGINX is up and running
kubectl get pods --namespace alfresco

NAME                                                   READY   STATUS    RESTARTS   AGE
acs-ingress-ingress-nginx-controller-5647c976f-f7b6q   1/1     Running   0          98m

# Verify expose localhost:80
kubectl get svc --namespace alfresco

NAME                                             TYPE           CLUSTER-IP     EXTERNAL-IP     PORT(S)                      AGE
acs-ingress-ingress-nginx-controller-admission   ClusterIP      10.43.42.230   <none>          443/TCP                      98m
acs-ingress-ingress-nginx-controller             LoadBalancer   10.43.90.117   192.168.29.69   80:31363/TCP,443:30980/TCP   98m
```

### ACS

This repository offers you the option to either deploy a system using stable released artifacts or the latest in-progress development artifacts.

To use a released version of the Helm chart add the stable chart repository:

```bash
helm repo add alfresco https://kubernetes-charts.alfresco.com/stable
helm repo update
```

> #### Community

Reviewer 2: just for consistency with the next chapter
```suggestion
#### Community localhost deployment
```

> To install the latest version of Community we need to use the [community_values.yaml file](../../helm/alfresco-content-services). Once downloaded, execute the following to initiate the deployment.

Reviewer 2: ```suggestion
To install the latest version of Community we need to download the [community_values.yaml file](../../helm/alfresco-content-services). Once downloaded, execute the following to initiate the deployment.
```

#### Enterprise localhost deployment

> To configure credentials for accessing the Alfresco Enterprise registry, please review the information provided in the [registry authentication](registry-authentication.md)

Reviewer 2: Explain briefly why we need it:

```suggestion
ACS enterprise version needs to pull container images from private image repositories.
To configure credentials for accessing the Alfresco Enterprise registry, please review the information provided in the [registry authentication](registry-authentication.md)
```

  --namespace alfresco
```

> > NOTE: The command will wait until the deployment is ready so please be patient. See below for [troubleshooting](./rancher-desktop-deployment.md#troubleshooting) tips.

Reviewer: ```suggestion
> NOTE: The command will wait until the deployment is ready so please be patient. See below for [troubleshooting](#troubleshooting) tips.
```

  --namespace alfresco
```

> NOTE: The command will wait until the deployment is ready so please be patient. See below for [troubleshooting](#troubleshooting) tips.

> The provided command installs the most current version of ACS Enterprise.

Reviewer 2: ```suggestion
The `helm` command above installs the most current released version of ACS Enterprise.
```

## Troubleshooting

> In the event of a deployment failure, it is important to recognize that resource constraints are a common underlying cause. For further insights and guidance, please refer to the dedicated sections below for detailed information on resource-related issues. Additionally, you can find more comprehensive troubleshooting advice in the [Helm Troubleshooting section](./README.md#Troubleshooting)

Reviewer 2: sounds redundant

```suggestion
In the event of a deployment failure, it is important to recognize that resource constraints are a common underlying cause. For further insights and guidance. Additionally, you can find more comprehensive troubleshooting advice in the [Helm Troubleshooting section](./README.md#Troubleshooting)
```
